server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/fileformats/meta/StarTreeMetadataWriter.java

```diff
@@ -128,13 +127,11 @@ private static void writeMeta(
         metaOut.writeVInt(numNodes);
 
         // number of dimensions
-        // TODO: Revisit the number of dimensions for timestamps (as we will split timestamp into min, hour, etc.)
-        metaOut.writeVInt(starTreeField.getDimensionsOrder().size());
+        metaOut.writeVInt(starTreeField.getDimensionNames().size());
```

Comment:
One thing to note, which we can maybe add as an comment too. 
All the dimension names that we are writing, they are going to be read as separate read dimensions

Reply:
since star tree document is based on dimensionNames, its self explanatory i feel.